com.chenyifaer.FaFaRunner.yml

```diff
@@ -0,0 +1,56 @@
+id: com.chenyifaer.FaFaRunner
+
+# https://docs.flatpak.org/en/latest/available-runtimes.html
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: fafarunner
+
+# These are the permissions that the application needs
+# Read more about finishing here: https://docs.flatpak.org/en/latest/manifests.html#finishing
+# https://docs.flatpak.org/en/latest/sandbox-permissions.html
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --device=dri
+  - --share=network
+
+modules:
+  - name: fafarunner
+    buildsystem: simple
+    build-commands:
+      - bsdtar -Oxf fafarunner.deb data.tar.zst | bsdtar -xf -
+      - install -Dm755 usr/share/fafarunner/fafarunner /app/bin/fafarunner
+      - cp -r usr/share/fafarunner/data usr/share/fafarunner/lib /app/bin/
+      - sed -i -e '/^Version=/d' -e 's|Icon=fafarunner|Icon=com.chenyifaer.FaFaRunner|' usr/share/applications/fafarunner.desktop
+      - install -Dm644 usr/share/applications/fafarunner.desktop /app/share/applications/com.chenyifaer.FaFaRunner.desktop
+      - install -Dm644 usr/share/icons/hicolor/128x128/apps/fafarunner.png /app/share/icons/hicolor/128x128/apps/com.chenyifaer.FaFaRunner.png
+      - install -Dm644 usr/share/icons/hicolor/256x256/apps/fafarunner.png /app/share/icons/hicolor/256x256/apps/com.chenyifaer.FaFaRunner.png
+      - sed -i -e 's|<id>fafarunner</id>|<id>com.chenyifaer.FaFaRunner</id>|' -e 's|fafarunner.desktop|com.chenyifaer.FaFaRunner.desktop|' usr/share/metainfo/fafarunner.metainfo.xml
```

Comment:
don't use sed. It doesn't have a failure mode.

Note that if you keep the desktop file with its original name and use `rename-desktop-file` in the manifest, then it's done for you. Same with the appstream file using `rename-appdata-file` and the icons using `rename-icon`.

Reply:
Sorry, I don't understand

Reply:
The manifest file is universal. Will it have any impact on other Linux distribution packages?

Please provide an example.

Reply:
> The manifest file is universal. Will it have any impact on other Linux distribution packages?

No `fafarunner` is an incorrect ID. It needs to be rDNS. It'd be better if you directly fix these issues in the .deb.

Add the old id `fafarunner` in the `provides` tag of metainfo and it should be fine:

```
<provides>
  <id>fafarunner</id>
</provides>

Reply:
Maybe this?

```
<id>com.chenyifaer.FaFaRunner</id>
<provides>
  <id>fafarunner</id>
</provides>
```

Reply:
Yes

Reply:
Can I understand it this way? I added these tags, and what happens at runtime is:
File rename
1. `fafarunner.png` -> `com.chenyifaer.FaFaRunner.png`
2. `fafarunner.desktop` -> `com.chenyifaer.FaFaRunner.desktop`
3. `fafarunner.metainfo.xml` -> `com.chenyifaer.FaFaRunner.metainfo.xml`

File content changes
1. The value of `Icon` in the `fafarunner.desktop` file is updated from `fafarunner` to `com.chenyifaer.FaFaRunner`
2. The value of `id` in the `fafarunner.metainfo.xml` file is updated from `fafarunner.desktop` to `com.chenyifaer.FaFaRunner.desktop`
3. The value of `launchable` in the `fafarunner.metainfo.xml` file is updated from `fafarunner.desktop` to `com.chenyifaer.FaFaRunner.desktop`

Reply:
In addition, the `.desktop` file is automatically generated by the packaging tool, and it contains some wrong fields. I have to use sed to remove these wrong fields.

Reply:
That's what `desktop-file-edit` is for. Not `sed`.

Reply:
Can `desktop-file-edit` automatically remove invalid fields in `.desktop` files?

Reply:
I think it might be more suitable for me to move the `.desktop` and `.metainfo.xml` files to this repository.

Reply:
No.

Reply:
I just need to change
```
<id>fafarunner</id>
```

to
```
<id>com.chenyifaer.FaFaRunner</id>
<provides>
  <id>fafarunner</id>
</provides>
```

and then no other changes are needed?